volplugin/utils.go

```diff
@@ -164,3 +189,26 @@ func splitPath(name string) (string, string, error) {
 func joinPath(tenant, name string) string {
 	return strings.Join([]string{tenant, name}, ".")
 }
+
+func addStopChan(name string) chan struct{} {
+	mountMutex.Lock()
+	stopChan := make(chan struct{}, 1)
+
+	if sc, ok := mountStopChans[name]; ok {
```

Comment:
Ah, this is some confusing logic I should clean up or comment.

Basically, if we already have a stop channel there we want to send a message JUST IN CASE to clean up the goroutine before we replace it with the new channel and goroutine.